### ISSUE TITLE
USB CDC 4G Module doesn't work with SIM7600E-H | 'process command timeout'

### ISSUE THREAD

**Issue by leofabri** [espressif/esp-iot-solution#161]
I'm experimenting with the USB Host feature on my ESP32S3 (DevKitC-1 v1.0) and in particular, I'm trying to connect my SIM7600E-H, an LTE Cat.4 modem. In "theory", this should be similar to the SIM7600C1 listed in the [example](https://github.com/espressif/esp-iot-solution/tree/usb/add_usb_solutions/examples/usb/host/usb_cdc_4g_module), but I can't get it to work.

What I tried:

- I have connected a stable power supply to power up the SIMCom Modem. This way, I can exclude problems with the power source;
- I have connected the RST pin as it is marked as **Necessary** in the documentation;
- I bought a USB 1.1 Termination from [here](aliexpress.com/item/1005001900423145.html), connected to a Linux machine, and issued `lsusb -v`, the result was the following:

```
Bus 001 Device 010: ID 1e0e:9001 Qualcomm / Option SimTech, Incorporated
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1e0e Qualcomm / Option
  idProduct          0x9001 
  bcdDevice            3.18
  iManufacturer           1 SimTech, Incorporated
  iProduct                2 SimTech, Incorporated
  iSerial                 3 0123456789ABCDEF
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0102
    bNumInterfaces          6
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass    255 Vendor Specific Subclass
      bInterfaceProtocol    255 Vendor Specific Protocol
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
      ** UNRECOGNIZED:  05 24 00 10 01
      ** UNRECOGNIZED:  05 24 01 00 00
      ** UNRECOGNIZED:  04 24 02 02
      ** UNRECOGNIZED:  05 24 06 00 00
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval              32
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
      ** UNRECOGNIZED:  05 24 00 10 01
      ** UNRECOGNIZED:  05 24 01 00 00
      ** UNRECOGNIZED:  04 24 02 02
      ** UNRECOGNIZED:  05 24 06 00 00
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x85  EP 5 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval              32
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x84  EP 4 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x03  EP 3 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        3
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
      ** UNRECOGNIZED:  05 24 00 10 01
      ** UNRECOGNIZED:  05 24 01 00 00
      ** UNRECOGNIZED:  04 24 02 02
      ** UNRECOGNIZED:  05 24 06 00 00
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x87  EP 7 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval              32
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x86  EP 6 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x04  EP 4 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        4
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0 
      bInterfaceProtocol      0 
      iInterface              0 
      ** UNRECOGNIZED:  05 24 00 10 01
      ** UNRECOGNIZED:  05 24 01 00 00
      ** UNRECOGNIZED:  04 24 02 02
      ** UNRECOGNIZED:  05 24 06 00 00
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x89  EP 9 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x000a  1x 10 bytes
        bInterval              32
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x88  EP 8 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x05  EP 5 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        5
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass    255 Vendor Specific Subclass
      bInterfaceProtocol    255 Vendor Specific Protocol
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x8b  EP 11 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval              32
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x8a  EP 10 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x06  EP 6 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
Device Qualifier (for other device speed):
  bLength                10
  bDescriptorType         6
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  bNumConfigurations      1
can't get debug descriptor: Resource temporarily unavailable
Device Status:     0x0000
  (Bus Powered)
```
### Notes and considerations:
I don't know exactly which one of those addresses is the USB_OUT_EP_ADDR and USB_IN_EP_ADDR, however, after seeing what the configuration looks like on the example, I think the pair should be this one:

**_CONFIG_MODEM_USB_OUT_EP_ADDR=0x01
CONFIG_MODEM_USB_IN_EP_ADDR=0x81_**

The SIM7600C1 seems to be using `0x0a` as `USB_OUT_EP_ADDR`. I did try to use that and, as expected, it didn't work. 

---

### Console output

This is what I'm getting when running the program:
```
W (373) main: Force reset 4g board
I (377) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
I (3893) main: ====================================
I (3893) main:      ESP 4G Cat.1 Wi-Fi Router
I (3893) main: ====================================
I (3948) USB_HCDC: usb driver install succeed
I (3948) USB_HCDC: Waitting Device Connection
I (3978) USB_HCDC: USB Port=1 init succeed
I (3978) USB_HCDC: Waitting USB Connection
I (3978) USB_HCDC: Port power: ON
I (4231) USB_HCDC: line 263 HCD_PORT_EVENT_CONNECTION
I (4231) USB_HCDC: Resetting Port
I (4291) USB_HCDC: Port reset succeed
I (4291) USB_HCDC: Getting Port Speed
I (4291) USB_HCDC: Port speed = 1
I (4293) USB_HCDC: USB Speed: full-speed
I (4297) USB_HCDC: Set Device Addr = 1
I (4303) USB_HCDC: Set Device Addr Done
I (4306) USB_HCDC: Set Device Configuration = 1
I (4314) USB_HCDC: Set Device Configuration Done
I (4317) USB_HCDC: Creating bulk in pipe
I (4321) USB_HCDC: Creating bulk out pipe
I (4328) USB_HCDC: Device Connected
I (4330) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
E (4840) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
 response timeoutm_dce: esp_modem_dce_generic_command(31): Command:AT
I (4845) modem_board: Current timeouts: 1 and errors: 0
E (9851) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
W (9851) esp_modem_dce: esp_modem_dce_generic_command(31): Command:+++ response timeout
E (10356) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
 response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
E (10861) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
 response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
I (10866) modem_board: Current timeouts: 2 and errors: 0
I (10872) modem_board: Restart to connect modem........
I (10878) modem_board: modem_board_reset!
E (16583) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
 response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
I (16588) modem_board: Current timeouts: 3 and errors: 0
I (16594) modem_board: modem_board_power_down!
E (17099) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
 response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
```

Basically, the modem doesn't seem to be replying to my AT commands. I can't verify if the problem is hardware or software. 
I made sure with an oscilloscope that the reset pin would work properly and it does look like so.

---

### My Config
```
#
# ESP-MODEM
#
# CONFIG_MODEM_TARGET_ML302_DNLM is not set
# CONFIG_MODEM_TARGET_AIR724UG_NFM is not set
# CONFIG_MODEM_TARGET_EC600N_CNLA_N05 is not set
# CONFIG_MODEM_TARGET_EC600N_CNLC_N06 is not set
# CONFIG_MODEM_TARGET_A7600C1 is not set
CONFIG_MODEM_TARGET_USER=y

#
# USB CDC endpoint address config
#
CONFIG_MODEM_USB_OUT_EP_ADDR=0x01
CONFIG_MODEM_USB_IN_EP_ADDR=0x81
# end of USB CDC endpoint address config

# CONFIG_MODEM_LEGACY_API is not set
# end of ESP-MODEM
# end of Component config

```

I can't get this sorted out.

**Comment by leofabri** [comment]
**Update:** After looking re-reading this issue: https://github.com/espressif/esp-iot-solution/issues/121#issuecomment-900775754 I think understood how to use the output of `lsusb -v`

**I'm going to mention how it's done for future reference.**
Take a look at the following lines (extracted from my first comment, here: https://github.com/espressif/esp-iot-solution/issues/161#issue-1113807950):
```Shell
bEndpointAddress 0x81 EP 1 IN
bEndpointAddress 0x01 EP 1 OUT

bEndpointAddress 0x83 EP 3 IN // Note: there's no OUT for this

bEndpointAddress 0x82 EP 2 IN
bEndpointAddress 0x02 EP 2 OUT

bEndpointAddress 0x85 EP 5 IN // Note: there's no OUT for this

bEndpointAddress 0x84 EP 4 IN
bEndpointAddress 0x03 EP 3 OUT

bEndpointAddress 0x87 EP 7 IN // Note: there's no OUT for this

bEndpointAddress 0x86 EP 6 IN
bEndpointAddress 0x04 EP 4 OUT

bEndpointAddress 0x89 EP 9 IN // Note: there's no OUT for this

bEndpointAddress 0x88 EP 8 IN
bEndpointAddress 0x05 EP 5 OUT

bEndpointAddress 0x8b EP 11 IN // Note: there's no OUT for this

bEndpointAddress 0x8a EP 10 IN
bEndpointAddress 0x06 EP 6 OUT
```

  

### After filtering out the INs without an OUT, I got the following pairs:

  

```
bEndpointAddress 0x81 EP 1 IN
bEndpointAddress 0x01 EP 1 OUT

bEndpointAddress 0x82 EP 2 IN
bEndpointAddress 0x02 EP 2 OUT

bEndpointAddress 0x84 EP 4 IN
bEndpointAddress 0x03 EP 3 OUT

bEndpointAddress 0x86 EP 6 IN
bEndpointAddress 0x04 EP 4 OUT

bEndpointAddress 0x88 EP 8 IN
bEndpointAddress 0x05 EP 5 OUT

bEndpointAddress 0x8a EP 10 IN
bEndpointAddress 0x06 EP 6 OUT
```

## Results when trying each EP, one by one
<br>
  

> ## EP(s): **[IN 0x81, OUT 0x01]** ❌
> ```Shell
> I (4330) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
> E (4840) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutm_dce: esp_modem_dce_generic_command(31): Command:AT
> I (4845) modem_board: Current timeouts: 1 and errors: 0
> E (9851) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
> W (9851) esp_modem_dce: esp_modem_dce_generic_command(31): Command:+++ response timeout
> E (10356) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
> E (10861) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
> I (10866) modem_board: Current timeouts: 2 and errors: 0
> I (10872) modem_board: Restart to connect modem........
> I (10878) modem_board: modem_board_reset!
> ```
___
> ## EP(s): **[IN 0x82, OUT 0x02]** ❌
> ```Shell
> I (4335) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
> E (4523) esp-modem-dte: esp_dte_handle_line(88): handle line failed
> E (4845) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutm_dce: esp_modem_dce_generic_command(31): Command:AT
> I (4850) modem_board: Current timeouts: 1 and errors: 0
> E (5524) esp-modem-dte: esp_dte_handle_line(88): handle line failed
> E (6523) esp-modem-dte: esp_dte_handle_line(88): handle line failed
> E (7526) esp-modem-dte: esp_dte_handle_line(88): handle line failed
> E (8524) esp-modem-dte: esp_dte_handle_line(88): handle line failed
> E (9523) esp-modem-dte: esp_dte_handle_line(88): handle line failed
> E (9856) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
> W (9856) esp_modem_dce: esp_modem_dce_generic_command(31): Command:+++ response timeout
> E (10361) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
> E (10523) esp-modem-dte: esp_dte_handle_line(88): handle line failed
> E (10866) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
> I (10871) modem_board: Current timeouts: 2 and errors: 0
> I (10877) modem_board: Restart to connect modem........
> I (10883) modem_board: modem_board_reset!
> I (11523) modem_board: xxxxxxxxxxxxxx Board ready after hard reset/power-cycle
> I (12522) modem_board: xxxxxxxxxxxxxx Board ready after hard reset/power-cycle
> I (13523) modem_board: xxxxxxxxxxxxxx Board ready after hard reset/power-cycle
> I (14523) modem_board: xxxxxxxxxxxxxx Board ready after hard reset/power-cycle
> I (15523) modem_board: xxxxxxxxxxxxxx Board ready after hard reset/power-cycle
> ```
___
> ## EP(s): **[IN 0x86, OUT 0x04]** ❌
> ```Shell
> I (8059) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
> E (8568) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutm_dce: esp_modem_dce_generic_command(31): Command:AT
> I (8573) modem_board: Current timeouts: 1 and errors: 0
> E (13579) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
> W (13579) esp_modem_dce: esp_modem_dce_generic_command(31): Command:+++ response timeout
> E (14084) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
> E (14589) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
> I (14594) modem_board: Current timeouts: 2 and errors: 0
> I (14600) modem_board: Restart to connect modem........
> I (14606) modem_board: modem_board_reset!
> I (14976) modem_board: xxxxxxxxxxxxxx Board ready after hard reset/power-cycle
> I (15195) modem_board: xxxxxxxxxxxxxx Board ready after hard reset/power-cycle
> I (15511) modem_board: xxxxxxxxxxxxxx Board ready after hard reset/power-cycle
> I (16481) modem_board: Board ready after hard reset/power-cycle
> I (21857) modem_board: Modem PPP Started
> W (21857) modem_board: PPP netif event = 259
> W (21857) modem_board: PPP netif event = 262
> W (21862) modem_board: PPP netif event = 263
> W (21865) modem_board: PPP netif event = 265
> ```
___
> ## EP(s): **[IN 0x88, OUT 0x05]** & **[IN 0x08a, OUT 0x06]** ❌
> ```Shell
> I (4335) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
> E (4845) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutm_dce: esp_modem_dce_generic_command(31): Command:AT
> I (4850) modem_board: Current timeouts: 1 and errors: 0
> E (9856) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
> W (9856) esp_modem_dce: esp_modem_dce_generic_command(31): Command:+++ response timeout
> E (10361) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
> E (10866) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutem_dce: esp_modem_dce_generic_command(31): Command:AT
> I (10871) modem_board: Current timeouts: 2 and errors: 0
> I (10877) modem_board: Restart to connect modem........
> I (10883) modem_board: modem_board_reset!
> ```
___
> ## EP(s): **IN 0x84, OUT 0x03** ✅ This is the right one.
> ```Shell
> I (6170) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0
> E (6680) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
>  response timeoutm_dce: esp_modem_dce_generic_command(31): Command:AT
> I (6685) modem_board: Current timeouts: 1 and errors: 0
> E (11691) esp-modem-dte: esp_modem_dte_send_cmd(196): process command timeout
> W (11691) esp_modem_dce: esp_modem_dce_generic_command(31): Command:+++ response timeout
> I (14147) modem_board: Modem PPP Started
> W (14147) modem_board: PPP netif event = 259
> W (14147) modem_board: PPP netif event = 262
> W (14153) modem_board: PPP netif event = 263
> W (14155) modem_board: PPP netif event = 265
> I (14171) esp-netif_lwip-ppp: Connected
> I (14171) esp-netif_lwip-ppp: Name Server1: <some_ip>
> I (14172) esp-netif_lwip-ppp: Name Server2: <some_ip>
> I (14177) modem_board: IP event! 6
> I (14181) modem_board: Modem Connected to PPP Server
> I (14187) modem_board: ppp ip: <some_ip>, mask: 255.255.255.255, gw: <some_ip>
> I (14195) modem_board: Main DNS: <some_dns_ip>
> I (14201) modem_board: Backup DNS: <some_dns_ip>
> W (14206) modem_board: PPP netif event = 266
> I (14211) main: ap dns addr(auto): <some_dns_ip>
> I (14216) pp: pp rom version: e7ae62f
> I (14221) net80211: net80211 rom version: e7ae62f
> I (14227) wifi:wifi driver task: 3fcf7cb0, prio:23, stack:6656, core=0
> I (14232) system_api: Base MAC address is not set
> I (14238) system_api: read default base MAC address from EFUSE
> I (14248) wifi:wifi firmware version: 0d44bbe
> I (14248) wifi:wifi certification version: v7.0
> I (14252) wifi:config NVS flash: enabled
> I (14256) wifi:config nano formating: disabled
> I (14260) wifi:Init data frame dynamic rx buffer num: 32
> I (14265) wifi:Init management frame dynamic rx buffer num: 32
> I (14271) wifi:Init management short buffer num: 32
> I (14275) wifi:Init dynamic tx buffer num: 32
> I (14280) wifi:Init static tx FG buffer num: 2
> I (14284) wifi:Init static rx buffer size: 1600
> I (14288) wifi:Init static rx buffer num: 10
> I (14292) wifi:Init dynamic rx buffer num: 32
> I (14296) wifi_init: tcpip mbox: 32
> I (14300) wifi_init: udp mbox: 6
> I (14304) wifi_init: tcp mbox: 6
> I (14308) wifi_init: tcp tx win: 5744
> I (14312) wifi_init: tcp rx win: 5744
> I (14317) wifi_init: tcp mss: 1440
> I (14321) wifi_init: WiFi IRAM OP enabled
> I (14325) wifi_init: WiFi RX IRAM OP enabled
> I (14330) wifi_init: LWIP IRAM OP enabled
> I (14335) phy_init: phy_version 302,3ddfab3,Dec  3 2021,14:39:12
> I (14383) wifi:mode : softAP (<mac_address>)
> I (14385) wifi:Total power save buffer number: 16
> I (14385) wifi:Init max length of beacon: 752/752
> I (14386) wifi:Init max length of beacon: 752/752
> I (14395) wifi:Total power save buffer number: 16
> I (14395) modem_wifi: softap ssid: esp_4g_router password: 12345678
> I (14401) modem_wifi: NAT is enabled
> I (14406) main: Task dump
> 
> I (14409) main: Load    Stack left      Name    PRI
> 
> I (14414) main: 1.11    1092    cdc-data        6
> 
> I (14418) main: 0.99    1368    main    1
> 
> I (14423) main: 95.75   904     IDLE    0
> 
> I (14427) main: 94.58   900     IDLE    0
> 
> I (14431) main: 0.32    1012    usb_proc        7
> 
> I (14436) main: 0.70    968     usb_data_recv   8
> 
> I (14441) main: 0.07    2356    tiT     18
> 
> I (14446) main: 0.00    1468    Tmr Svc         1
> 
> I (14450) main: 2.73    864     ipc1    24
> 
> I (14454) main: 0.00    3444    esp_timer       22
> 
> I (14459) main: 0.31    700     sys_evt         20
> 
> I (14464) main: 0.86    4204    wifi    23
> 
> I (14468) main: 2.57    908     ipc0    24
> 
> I (14473) main: Free heap=213848 Free mini=208896 bigst=213848, internal=213540 bigst=208896
> I (14482) main: ..............
> 
> I (14486) USB_HCDC: USBH CDC Transfer Buffer Dump:
> I (14491) USB_HCDC: usb transfer Buffer size, out = 1536, in = 1536
> I (14498) USB_HCDC: usb transfer Max packet size, out = 83, in = 56
> 
> I (14505) USB_HCDC: USBH CDC Ringbuffer Dump:
> I (14510) USB_HCDC: usb ringbuffer size, out = 15360, in = 15360
> I (14517) USB_HCDC: usb ringbuffer load, out = 0, in = 0
> I (14523) USB_HCDC: usb ringbuffer High water mark, out = 83, in = 56
> ```
___

# Solution:

For the SIM7600E-H, the right configuration is:
```
CONFIG_MODEM_USB_OUT_EP_ADDR=0x03
CONFIG_MODEM_USB_IN_EP_ADDR=0x84
```

I think this can be closed now.
